pypeit/core/flexure.py

```diff
@@ -303,147 +305,226 @@ def spec_flex_shift(obj_skyspec, arx_skyspec, arx_lines, mxshft=20):
+                    waveimg_flex_corr[slitmask == slit_spat] = \
+                        specobj.SpecObj.flexure_interp(fdict['shift'], sky_wave, wavein=waveimg[slitmask == slit_spat])
+        else:
+            i_slitord = slitord[islit]
+            indx = specobjs.slitorder_indices(i_slitord)
+            this_specobjs = specobjs[indx]
+            # Loop through objects
+            for ss, sobj in enumerate(this_specobjs):
+                if sobj is None:
+                    continue
+                if sobj['BOX_WAVE'] is None: #len(specobj._data.keys()) == 1:  # Nothing extracted; only the trace exists
+                    continue
+                msgs.info("Working on flexure for object # {:d}".format(sobj.OBJID) + "in slit # {:d}".format(islit))
+
+                # Using boxcar
+                sky_wave = sobj.BOX_WAVE
+                sky_flux = sobj.BOX_COUNTS_SKY
+
+                # Generate 1D spectrum for object
+                obj_sky = xspectrum1d.XSpectrum1D.from_tuple((sky_wave, sky_flux))
+
+                # Calculate the shift
+                fdict = spec_flex_shift(obj_sky, sky_spectrum, sky_lines, mxshft=mxshft)
+                punt = False
+                if fdict is None:
+                    msgs.warn("Flexure shift calculation failed for this spectrum.")
+                    if sv_fdict is not None:
+                        msgs.warn("Will used saved estimate from a previous slit/object")
+                        fdict = copy.deepcopy(sv_fdict)
+                    else:
+                        # One does not exist yet
+                        # Save it for later
+                        return_later_sobjs.append([islit, ss])
+                        punt = True
                 else:
-                    # One does not exist yet
-                    # Save it for later
-                    return_later_sobjs.append([islit, ss])
-                    punt = True
-            else:
-                sv_fdict = copy.deepcopy(fdict)
-
-            # Punt?
-            if punt:
-                break
-
-            # Interpolate
-            new_sky = specobj.flexure_interp(sky_wave, fdict)
-            # Update dict
-            for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec']:
-                flex_dict[key].append(fdict[key])
-            flex_dict['sky_spec'].append(new_sky)
-
+                    sv_fdict = copy.deepcopy(fdict)
+
+                # Punt?
+                if punt:
+                    break
+
+                # Interpolate
+                new_sky = sobj.apply_spectral_flexure(fdict)
+                if slitShift is not None:
+                    sobj.update_flex_shift(slitShift[islit])
+                # Update dict
+                for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec']:
+                    flex_dict[key].append(fdict[key])
+                flex_dict['sky_spec'].append(new_sky)
+
```

Comment:
It looks like all the code below is defunct, since it just throws an error. @profxj do you want to comment.